src/Common/tests/Cryptography/AlgorithmImplementations/RSA/ImportExport.cs

```diff
@@ -189,12 +242,28 @@ internal static void AssertKeyEquals(ref RSAParameters expected, ref RSAParamete
         {
             Assert.Equal(expected.Modulus, actual.Modulus);
             Assert.Equal(expected.Exponent, actual.Exponent);
-            Assert.Equal(expected.D, actual.D);
+            
             Assert.Equal(expected.P, actual.P);
             Assert.Equal(expected.DP, actual.DP);
             Assert.Equal(expected.Q, actual.Q);
             Assert.Equal(expected.DQ, actual.DQ);
             Assert.Equal(expected.InverseQ, actual.InverseQ);
+
+            if (expected.D == null)
+            {
+                Assert.Null(actual.D);
+            }
+            else
+            {
+                Assert.NotNull(actual.D);
+
+                // If the value matched expected, take that as valid and shortcut the math.
+                // If it didn't, we'll test that the value is at least legal.
+                if (!expected.D.SequenceEqual(actual.D))
```

Comment:
I had a Console.WriteLine in `VerifyDValue` during development, so I know that the CNG tests got in there exactly twice, which is what I'd expect.